clients/src/main/java/org/apache/kafka/common/telemetry/ClientTelemetry.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.telemetry;
+
+import org.apache.kafka.common.metrics.MetricsContext;
+import org.apache.kafka.common.metrics.MetricsReporter;
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.Optional;
+
+/**
+ * A {@link MetricsReporter} may implement this interface to indicate support for collecting client
```

Comment:
If this interface is only useful if it's implementing class is a `MetricsReporter`, why not have this interface extend from `MetricsReporter`? I'm sure there's a good reason that I'm missing.

Reply:
Just repeating the same I mentioned on other comment.

I was in an impression that we want Future MetricsReporter to support either client or server side mapping. And that's why this change existed in v1 of the code as well.

Looking at it again seems I am creating fat interfaces and doesn't make much sense to create common ClientTelemetry interface with optional client and server behaviour rather split them. And let the implementation of MetricsReporter decide which client or server side additional provider it wants to support. I ll change it.

Reply:
> to support either client or server side mapping
What does this mean?

My understanding was that `ClientTelemetry` is something that is only used client side (for this case, I tend to agree with Kirk -- if we really go this route, we need to update the KIP accordingly).

Reply:
I have simplified it to have what's in the KIP. Do you think the right place for it should be `server-common/org/apache/kafka/metrics/`? Is there any other better place for these i.e ClienTelemetryReceiver.java, ClientTelemetry.java and ClientTelemetryPayload.java which will be specifically for broker.

Reply:
Thinking again I think it should be good to layout Client Specific Interfaces which broker might implement if only wants to support client telemetry in client packages. We shouldn't bloat server packages with interfaces which an external dependency needs to implement. Please let me know your thoughts.

Reply:
My view is that this is now in the correct place in the source tree. It's a mixin for `MetricsReporter` and that's in `clients/src/main/java/org/apache/kafka/common/metrics`. So, I think this is correct.